src/status_im2/contexts/chat/images_horizontal/view.cljs

```diff
@@ -0,0 +1,67 @@
+(ns status-im2.contexts.chat.images-horizontal.view
+  (:require [quo2.core :as quo]
+            [quo2.foundations.colors :as colors]
+            [react-native.core :as rn]
+            [react-native.fast-image :as fast-image]
+            [utils.re-frame :as rf]
+            [react-native.safe-area :as safe-area]
+            [reagent.core :as reagent]
+            [oops.core :as oops]
+            [status-im2.contexts.chat.images-horizontal.style :as style]))
+
+(defn image
+  [message]
+  (let [shared-element-id (rf/sub [:shared-element-id])
+        window-width      (:width (rn/get-window))
+        height            (* (:image-height message) (/ window-width (:image-width message)))]
+    [fast-image/fast-image
+     {:source    {:uri (:image (:content message))}
+      :style     {:width         window-width
+                  :height        height
+                  :border-radius 12}
+      :native-ID (when (= shared-element-id (:message-id message)) :shared-element)}]))
+
+(defn get-item-layout
+  [_ index]
+  #js {:length (:width (rn/get-window)) :offset (* (:width (rn/get-window)) index) :index index})
+
+(defn on-viewable-items-changed
+  [e]
+  (rf/dispatch [:chat.ui/update-shared-element-id
+                (:message-id (oops/oget (first (oops/oget e "changed")) "item"))]))
+
+
+(defn images-horizontal
+  []
+  (let [{:keys [messages index]} (rf/sub [:get-screen-params])
+        data                     (reagent/atom [(nth messages index)])
+        flat-list-ref            (atom nil)]
+    (reset! data messages)
+    (.then (new js/Promise (fn [resolve] (js/setTimeout resolve 0)))
```

Comment:
@ilmotta sorry for not writing an explanation.

So here we have the initial value of `data` as the image that was pressed (and not the whole album) in order for the transition animation to execute properly, otherwise, it would animate towards outside the screen (even if we have `initialScrollIndex` set). Then we need use setTimeout to enqueue `scrollToIndex` until the `data` has been updated. 
Upon further testing, I found the `Promise` to not be needed, and the `setTimeout` to be sufficient.

Reply:
Thank you @OmarBasem, it's clearer to me now.